marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -151,11 +151,28 @@ protected JsonNode determineIngestionEndpointParams(Map<String, String> params)
         }
 
         if (!endpointParams.hasNonNull("workUnit")) {
-            ObjectNode defaultWorkUnit = objectMapper.createObjectNode();
-            defaultWorkUnit.put("uriprefix", params.get("uriprefix")!=null ? params.get("uriprefix"):"");
+            defaultWorkUnit = objectMapper.createObjectNode();
+            defaultWorkUnit.put("uriprefix", params.get("uriprefix") != null ? params.get("uriprefix") : "");
```

Comment:
uriprefix should go into buildWorkUnitFromParams. Also, I'd just call that "buildDefaultWorkUnit" - it's clear that it's built from params because that's the sole argument.

Reply:
Added uriprefix to buildDefaultWorkUnit